src/actions/profileActions.js

```diff
@@ -0,0 +1,147 @@
+import Axios from 'axios';
+import Cookies from 'js-cookie';
+
+export const PROFILE_DETAILS = 'PROFILE DETAILS';
+export const profileDetails = details => (
+  {
+    type: PROFILE_DETAILS,
+    details,
+  }
+);
+
+export const UPDATING = 'updating profile details';
+export const updating = status => (
+  {
+    type: UPDATING,
+    updating: status,
+  }
+);
+
+export const ERRORS = 'failed to update profile details';
+export const errors = errorsDetails => (
+  {
+    type: ERRORS,
+    errors: errorsDetails,
+  }
+);
+
+export const UPDATEPROFILE = 'update profile detail status';
+export const updated = status => (
+  {
+    type: UPDATEPROFILE,
+    status,
+  }
+);
+
+const token = Cookies.get('access_token');
+const username = Cookies.get('username');
+export const getProfileDetails = () => (
+  dispatch => (
+    Axios.get(
+      `https://ah-technocrats.herokuapp.com/api/profiles/${username}/`,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Token ${token}`,
+        },
+      },
+    ).then(
+      (response) => {
+        dispatch(profileDetails(response.data.profile));
+      },
+      (error) => {
+        console.log('An error occured', error);
+      },
+    )
+  )
+);
+const updateErrorDispatch = (dispatch, error) => {
+  dispatch(updating(false));
+  dispatch(updated(false));
+  let err = error.response.data.errors;
+  if (err.username) {
+    err = { ...err.profile, username: [err.username] };
+    dispatch(errors(err));
+  } else if (err.email) {
+    err = { ...err.profile, email: [err.email] };
+    dispatch(errors(err));
+  } else {
+    dispatch(errors(err.profile));
+  }
+};
+
+export const FOLLOWING = 'users following';
+export const following = followingCount => (
+  {
+    type: FOLLOWING,
+    following: followingCount,
+  }
+);
+
+export const FOLLOWERS = 'user followers';
+export const followers = followersCount => ({
+  type: FOLLOWERS,
+  followers: followersCount,
+});
+
+const conf = ({
+  headers: {
+    'Content-Type': 'application/json',
+    Authorization: `Token ${token}`,
+  },
+  baseURL: `https://ah-technocrats.herokuapp.com/api/profiles/${username}/`,
+});
+
+const getData = (dispatch, endpoint) => Axios.get(endpoint, conf)
+  .then((response) => {
+    const whichDispatch = { following, followers };
+    let dataReturned;
+    if (endpoint === 'following') {
+      dataReturned = response.data.following.length;
+    } else if (endpoint === 'followers') {
+      dataReturned = response.data.followers.length;
+    }
+    dispatch(whichDispatch[endpoint](dataReturned));
+  }).catch((error) => {
+    console.log(error);
```

Comment:
Instead of printing the errors on the console, could you show the errors using a toast method or render it on the page. A normal user will not look at the look console when things go wrong.